### PR TITLE
on-core changes required for 2.0 workflows routes.

### PR DESCRIPTION
**Pull request by brianparry**
* Modifications to workflow store by @srinia6 and @abildw
* Add active() method to graph-object model.

### CODE REVIEW

File: lib/workflow/stores/mongo.js [hunk 1]
```diff
@@ -803,6 +820,20 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
     };
 
     /**
+     * Deletes a given task  by taskName from the taskobjects collection
+     * @param {String} injectableName - the injectable name for the desired task to be deleted
+     * @returns {Promise}
+     * @memberOf store
+     */
+    exports.deleteTaskByName = function(injectableName) {
+        var query = {
```
Comment (zyoung51) [inline lib/workflow/stores/mongo.js]: Should this have an assert like destroyGraphDefinition does?

File: spec/lib/workflow/stores/mongo-spec.js [hunk 2]
```diff
@@ -643,4 +644,35 @@ describe('Task Graph mongo store interface', function () {
...
+        return mongo.deleteTaskByName('testname')
+        .then(function() {
+            expect(waterline.taskdefinitions.destroy).to.have.been.calledOnce;
+            expect(waterline.taskdefinitions.destroy).to.have.been.calledWith(
+                { injectableName: 'testname' }
+            );
+        });
+    });
+
+    it('deleteTaskByName should fail on an invalid query(injectable name)', function() {
```
Comment (abildw) [inline spec/lib/workflow/stores/mongo-spec.js]: @srinia6 didn't we want to get rid of this negative test since it doesn't return the error properly?

Comment (srinia6) [inline spec/lib/workflow/stores/mongo-spec.js]: @abildw: your patch had that change, right?

**Comment by abildw** [comment]
made another commit with @srinia6 making sure we pushed the latest test and added in the assert @zyoung51 was asking about as well

**Comment by benbp** [comment]
:+1: but can we squash this down a little bit? No reason to have 10 commits for 67 lines changed.

**Comment by srinia6** [comment]
test this please

**Comment by yyscamper** [comment]
Agree with @benbp , :+1: after squash the commit.

**Comment by brianparry** [comment]
There was a bit of a rebasing mixup that resulted in all these extra commits.  @abildw is cleaning this up.

**Comment by srinia6** [comment]
test this please

**Comment by brianparry** [comment]
I opened https://github.com/RackHD/RackHD/issues/150 to track this pollers test failure.  Jenkins, test this please.

**Comment by abildw** [comment]
+1

**Pull request merged**